internetarchive/openlibrary
# Cookie-based banner preferences

Modifies our dismissible banners in the following ways:
1. "Hide banner" preferences are stored in cookies, not `User` preferences
2. Multiple banners can be dismissed at once (useful during periods when both a site banner and My Books page banner co-exist).
3. Logic to hide a banner has been moved to the banner template itself.

### Technical

Client-side code has be reorganized, and some functions have been renamed.  `id`s have been removed from the banner template in order to support multiple banners per page.

### Testing

1. Set an announcement and cookie name in `site/body.html`
2. While logged out, visit any page and ensure that the banner is present.  It should not have the "X".
3. Log into the site.  Ensure that the banner is present, and that it has an "X".
4. Click the "X".  Ensure that the banner is immediately hidden.
5. Refresh the page.  Ensure that the banner is still hidden.
6. Check your cookies, ensuring that the name is as expected.

### Screenshot


### Stakeholders
